flexmeasures/data/models/forecasting/residential/src/pipelines/base_pipeline.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import annotations
+
+import sys
+from datetime import datetime
+
+import pandas as pd
+from darts import TimeSeries
+from darts.dataprocessing.transformers import MissingValuesFiller
+from flexmeasures.data import db
+from flexmeasures.data.models.time_series import Sensor
+from flexmeasures.data.utils import get_data_source
+
+from ..exception import CustomException
+from ..logger import logging
+
+
+class BasePipeline:
```

Comment:
A short docstring to explain what this is would be helpful. Also for the subclasses.